pkg/controller/telemetry/metric.go

```diff
@@ -429,36 +476,124 @@ func buildPrincipal(workload *workloadapi.Workload) string {
 	return "-"
 }
 
-func buildWorkloadMetricsToPrometheus(data requestMetric, labels workloadMetricLabels) {
-	commonLabels := struct2map(labels)
-
-	if data.state == TCP_ESTABLISHED {
-		tcpConnectionOpenedInWorkload.With(commonLabels).Add(float64(1))
-	}
-	if data.state == TCP_CLOSTED {
-		tcpConnectionClosedInWorkload.With(commonLabels).Add(float64(1))
+func (m *MetricController) buildWorkloadMetricsToPrometheus(data requestMetric, labels workloadMetricLabels) {
+	_, ok := m.metricCache.WorkloadConnReceivedBytes[labels]
```

Comment:
I am a little confused, donot we need to lock it

Reply:
Not need lock it.
Getting `TcpProbeInfo` to `buildMetricToPrometheus` is step by step, there are no multiple goroutines.